### PR TITLE
Remove all uses of `SString::Printf()` accepting `WCHAR`

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This removes all uses of the `SString::Printf(const WCHAR* fmt, ...)` from the VM code base. It is another step toward removing the custom `wprintf` logic from the PAL.

/cc @jkotas @janvorli @elinor-fung @jkoritzinsky

### CODE REVIEW

File: src/coreclr/vm/typedesc.cpp
Comment (jkotas) [inline src/coreclr/vm/typedesc.cpp:227]:
```suggestion
            ssBuff.Append(W('!'));
```

File: src/coreclr/vm/nativelibrary.cpp
Comment (jkotas) [inline src/coreclr/vm/nativelibrary.cpp:676]: Where did the `"."` come from? It does not make sense.

File: src/coreclr/debug/di/eventredirectionpipeline.cpp
Comment (jkotas) [inline src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: Is this Windows specific, or just delaying the problem with `swprintf_s` till later?

File: src/coreclr/vm/nativelibrary.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/nativelibrary.cpp:676]: Oops. That is from a misreading of the format. Will revert to `W("")`

File: src/coreclr/debug/di/eventredirectionpipeline.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: This is windows specific. This code path is specifically for the strikeRS feature that was never ported to CoreCLR.

File: src/coreclr/vm/nativelibrary.cpp
Comment (jkotas) [inline src/coreclr/vm/nativelibrary.cpp:674]:
```suggestion
            if ((variations & NameVariations_Prefix) != 0)
```
?

File: src/coreclr/debug/di/eventredirectionpipeline.cpp
Comment (jkotas) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: Should this be deleted instead?

File: src/coreclr/vm/nativelibrary.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/nativelibrary.cpp:674]: Ugh. I'm sorry. This is not what I wanted to do. Blech. This is all wrong.

**Comment by AaronRobinsonMSFT** [comment]
CI is green. Please take another look. Thanks.

File: src/coreclr/binder/bindertracing.cpp
Comment (jkotas) [inline src/coreclr/binder/bindertracing.cpp:333]:
```suggestion
                        errorMsg.Printf("Resolution failed with HRESULT (%08x)", m_hr);
```
Anything wrong with doing the simple thing here?

File: src/coreclr/debug/di/eventredirectionpipeline.cpp
Comment (jkotas) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: @noahfalk Should this unused code be deleted?

File: src/coreclr/vm/dllimportcallback.cpp
Comment (jkotas) [inline src/coreclr/vm/dllimportcallback.cpp:316]:
```suggestion
        pMethodDesc->GetModule()->GetSimpleName(),
```

Comment (jkotas) [reply src/coreclr/vm/dllimportcallback.cpp:316]: (and delete `SString moduleName;`)

**Review by jkotas — APPROVED** [review]
LGTM, modulo comments

File: src/coreclr/binder/bindertracing.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/binder/bindertracing.cpp:333]: Nope. I'll move to that.

File: src/coreclr/utilcode/util_nodependencies.cpp
Comment (elinor-fung) [inline src/coreclr/utilcode/util_nodependencies.cpp:450]: Shouldn't this be `sprintf`?

File: src/coreclr/vm/clrex.cpp
Comment (elinor-fung) [inline src/coreclr/vm/clrex.cpp:1961]:
```suggestion
```

File: src/coreclr/vm/typedesc.cpp
Comment (janvorli) [inline src/coreclr/vm/typedesc.cpp:235]: A nit - looking at all the `FormatInteger` usages, they add quite a bit of churn to the code. I was wondering if adding `AppendInteger` method on the `SString` would be cleaner.

File: src/coreclr/utilcode/util_nodependencies.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/utilcode/util_nodependencies.cpp:450]: Yes it should.

File: src/coreclr/vm/typedesc.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/typedesc.cpp:235]: I had the same idea. I want to do this as well. My concern was the format. We have x, u, and d and expressing those is cumbersome at present. I want to take a hard look at `SString` in the future, see https://github.com/dotnet/runtime/issues/76581 for another example of annoyance. I can do that now or continue to focus on removing the printf logic, which I'd prefer. Once that is done `SString` is going to get an overhaul and `AppendInteger` is on the list.

File: src/coreclr/vm/clrex.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/clrex.cpp:1961]: Sad.

File: src/coreclr/vm/typedesc.cpp
Comment (janvorli) [reply src/coreclr/vm/typedesc.cpp:235]: Sounds good then.

File: src/coreclr/debug/di/eventredirectionpipeline.cpp
Comment (noahfalk) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: IMO, yeah we could delete this stuff. I am not aware of it getting any use for many years. @dotnet/dotnet-diag - Does anyone else have concerns about concerns about us deleting this?

@AaronRobinsonMSFT - I'd say give folks 24 hours to respond and if we don't have any objections consider that the green light to delete this code if you'd like.

Comment (hoyosjs) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp:111]: strikeRS is something @tommcdon was looking at, but I was of the same opinion. This is essentially old code that's just adding maintenance burden right now.

Comment (AaronRobinsonMSFT) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp]: I've created a separate commit that removes it and all uses for the redirection infrastructure. Please let me know if I should revert the commit.

Comment (tommcdon) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp]: No concerns from my end removing strikers support.  We can always revisit it if we decide to enable support in the future.

Comment (AaronRobinsonMSFT) [reply src/coreclr/debug/di/eventredirectionpipeline.cpp]: Thanks @tommcdon and @hoyosjs. Given your responses coupled with @noahfalk I am going to assume this is enough representation from the Diagnostics team that I can merge this in when green. Appreciate the quick responses.

File: src/coreclr/vm/clrex.cpp
Comment (elinor-fung) [inline src/coreclr/vm/clrex.cpp:1960]: Does the buffer size here and in the other uses of `FormatInteger` need `+1` for the terminating null?

**Review by elinor-fung — APPROVED** [review]

File: src/coreclr/vm/clrex.cpp
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/clrex.cpp:1960]: It doesn't for this one because it is included in `ARRAY_SIZE("#")`, but when it is one of the constants, it does. This is another justification for making a smart helper.

**Pull request merged**
